encoder/vaapiencoder_h264.cpp

```diff
@@ -77,6 +87,29 @@ static const H264LevelLimits LevelLimits[] = {
     {51, 983040, 2},
 };
 
+#define H264_MIN_TEMPORAL_GOP 8
+#define H264_MAX_TEMPORAL_LAYER_NUM 4
+
+static uint32_t TempIds[H264_MAX_TEMPORAL_LAYER_NUM][H264_MIN_TEMPORAL_GOP]
+    = { { 0, 0, 0, 0, 0, 0, 0, 0 },
+        { 0, 1, 0, 1, 0, 1, 0, 1 },
+        { 0, 2, 1, 2, 0, 2, 1, 2 },
+        { 0, 3, 2, 3, 1, 3, 2, 3 } };
+
+static const float LayerFps[H264_MAX_TEMPORAL_LAYER_NUM] = { 7.5, 15, 30, 60 };
+
+static uint32_t log2(uint32_t num)
```

Comment:
Hand writing things are error prone and low performance, if possbile, we had better use standard library if it have.